controllers/util/events.go

```diff
@@ -6,4 +6,7 @@ const (
 
 	// EventReasonValidationFailed is used when the StorageCluster spec validation fails
 	EventReasonValidationFailed = "FailedValidation"
+
+	// EventReasonUninstallFailed is used when the StorageCluster uninstall fails
+	EventReasonUninstallFailed = "FailedUninstall"
```

Comment:
I think the event is generated in two cases:
1. valid error where we know for sure that it won't progress
2. error where we are just waiting for the uninstall to progress in a sub component.
Unfortunately, you don't get to know the category in the caller.

Therefore, "FailedUninstall" might not be the best wording to use here. How about 
* UninstallInProgress
* UninstallPending

Reply:
I like UninstallPending, Updated the same